src/presign/participant.rs

```diff
@@ -299,10 +331,10 @@ impl ProtocolParticipant for PresignParticipant {
 }
 
 impl InnerProtocolParticipant for PresignParticipant {
-    type Context = SharedContext;
+    type Context = PresignContext;
 
     fn retrieve_context(&self) -> <Self as InnerProtocolParticipant>::Context {
-        SharedContext::collect(self)
+        PresignContext::collect(self.input().all_auxinfo_public[0].params(), self)
```

Comment:
I'm assuming here that the parameters are the same across parties. Otherwise I am not sure how the protocol would work.